src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMultiSignMessage.cs

```diff
@@ -0,0 +1,434 @@
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotReadable, nameof(detachedContent));
+
+            if (!detachedContent.CanSeek)
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotSeekable, nameof(detachedContent));
+
+            return SignCore(default, detachedContent, signer, protectedHeaders, unprotectedHeaders, associatedData, isDetached: true);
+        }
+
+        private static byte[] SignCore(ReadOnlySpan<byte> content, Stream? contentStream, CoseSigner signer, CoseHeaderMap? protectedHeaders, CoseHeaderMap? unprotectedHeaders, ReadOnlySpan<byte> associatedData, bool isDetached)
+        {
+            if (signer is null)
+                throw new ArgumentNullException(nameof(signer));
+
+            ValidateBeforeSign(signer, protectedHeaders, unprotectedHeaders, out int? algHeaderValueToSlip);
+
+            int expectedSize = ComputeEncodedSize(signer, protectedHeaders, unprotectedHeaders, algHeaderValueToSlip, content.Length, isDetached);
+            var buffer = new byte[expectedSize];
+
+            int bytesWritten = CreateCoseMultiSignMessage(content, contentStream, buffer, signer, protectedHeaders, unprotectedHeaders, associatedData, isDetached, algHeaderValueToSlip);
+            Debug.Assert(expectedSize == bytesWritten);
+
+            return buffer;
+        }
+
+        public static Task<byte[]> SignDetachedAsync(
+            Stream detachedContent,
+            CoseSigner signer,
+            CoseHeaderMap? protectedHeaders = null,
+            CoseHeaderMap? unprotectedHeaders = null,
+            ReadOnlyMemory<byte> associatedData = default,
+            CancellationToken cancellationToken = default)
+        {
+            if (detachedContent is null)
+                throw new ArgumentNullException(nameof(detachedContent));
+
+            if (signer is null)
+                throw new ArgumentNullException(nameof(signer));
+
+            if (!detachedContent.CanRead)
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotReadable, nameof(detachedContent));
+
+            if (!detachedContent.CanSeek)
+                throw new ArgumentException(SR.Sign1ArgumentStreamNotSeekable, nameof(detachedContent));
+
+            ValidateBeforeSign(signer, protectedHeaders, unprotectedHeaders, out int? algHeaderValueToSlip);
+
+            int expectedSize = ComputeEncodedSize(signer, protectedHeaders, unprotectedHeaders, algHeaderValueToSlip, contentLength: 0, isDetached: true);
+            return SignAsyncCore(expectedSize, detachedContent, signer, protectedHeaders, unprotectedHeaders, associatedData, cancellationToken, algHeaderValueToSlip);
+        }
+
+        private static async Task<byte[]> SignAsyncCore(int expectedSize, Stream content, CoseSigner signer, CoseHeaderMap? protectedHeaders, CoseHeaderMap? unprotectedHeaders, ReadOnlyMemory<byte> associatedData, CancellationToken cancellationToken, int? algHeaderValueToSlip)
```

Comment:
A lot of the functions in this file could use with being chopped across multiple lines.  This one and the next few are longer than my widescreen monitor, so should definitely be chopped.

```suggestion
        private static async Task<byte[]> SignAsyncCore(
            int expectedSize,
            Stream content,
            CoseSigner signer,
            CoseHeaderMap? protectedHeaders,
            CoseHeaderMap? unprotectedHeaders,
            ReadOnlyMemory<byte> associatedData,
            CancellationToken cancellationToken,
            int? algHeaderValueToSlip)
```

Oh, there was even a parameter after the cancellation token.  Literally wouldn't have guessed that.  (Consider flipping those for consistency)